ext/openssl/tests/openssl_csr_sign_string_serial.phpt

```diff
@@ -0,0 +1,65 @@
+--TEST--
+openssl_csr_sign() allows string serial
+--EXTENSIONS--
+openssl
+--FILE--
+<?php
+$config = __DIR__ . DIRECTORY_SEPARATOR . 'openssl.cnf';
+$config_arg = array('config' => $config);
+
+$dn = array(
+    "countryName" => "BR",
+    "stateOrProvinceName" => "Rio Grande do Sul",
+    "localityName" => "Porto Alegre",
+    "commonName" => "Henrique do N. Angelo",
+    "emailAddress" => "hnangelo@php.net"
+);
+
+$args = array(
+    "digest_alg" => "sha256",
+    "private_key_bits" => 2048,
+    "private_key_type" => OPENSSL_KEYTYPE_DSA,
+    "encrypt_key" => true,
+    "config" => $config
+);
+
+$privkey = openssl_pkey_new($config_arg);
+$csr = openssl_csr_new($dn, $privkey, $args);
+
+var_dump($cert1 = openssl_csr_sign($csr, null, $privkey, 365, $args, 1234567));
```

Comment:
Your change breaks code if this line looks like (number is a string - strict types disabled - default)

```
var_dump($cert1 = openssl_csr_sign($csr, null, $privkey, 365, $args, '1234567'));
```

Currently it returns the same result with weak types as it gets converted to string. With your change, you try to parse it as a hex and it results in error (just tried to be sure).